packages/snaps-controllers/src/services/webview/WebViewMessageStream.ts

```diff
@@ -0,0 +1,106 @@
+import type { PostMessageEvent } from '@metamask/post-message-stream';
+import { BasePostMessageStream } from '@metamask/post-message-stream';
+import { isValidStreamMessage } from '@metamask/post-message-stream/dist/utils';
+import { logError } from '@metamask/snaps-utils';
+import { assert, bytesToBase64, stringToBytes } from '@metamask/utils';
+
+// TBD
+export type WebViewInterface = {
+  injectJavaScript(js: string): void;
+  addEventListener(
+    type: 'message',
+    listener: (event: MessageEvent) => void,
+  ): void;
+  removeEventListener(
+    type: 'message',
+    listener: (event: MessageEvent) => void,
+  ): void;
```

Comment:
```suggestion
  registerMessageListener(
    listener: (event: MessageEvent) => void,
  ): void;
  unregisterMessageListener(
    listener: (event: MessageEvent) => void,
  ): void;
```

How do you feel about this?

In the mobile PR we throw away the type anyway and only support messages.

Reply:
Although we throw away it on mobile side, we make usage of addEventListener on getWebView. So how could that work if window [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) doesn't export such methods ?

Reply:
I'm not sure I follow. Where do we use this API besides in this class? I am only suggesting we change the WebView API that we control.